src/working-key/working-key.js

```diff
@@ -68,6 +98,36 @@ const resetIndices = (globalSpace, keyName) => {
     })
 }
 
+/**
+ * returns a deep copy of the existing key, usually used as a restore point later
+ */
+const copyWorkingKey = (globalSpace, keyName) => {
+  assertGlobalSpaceAndEngine('keyName', globalSpace, keyName)
+
+  const key = getWorkingKey(globalSpace, keyName)
+  if (!key) return
```

Comment:
I thought you didn't want to be clever with single line returns?

Reply:
for guard clauses single line conditionals and returns are appropriate (see https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html)